ha/monitor/k8s/monitor.py

```diff
@@ -51,36 +51,54 @@ def __init__(self, wait_time=10):
 
             self.monitors = []
+            monitor_args = {}
 
             # event output in pretty format
-            kwargs = {K8SClientConst.PRETTY : True}
+            monitor_args['watch_args'] = {K8SClientConst.PRETTY : True}
 
-            # Seting a timeout value, 'timout_seconds', for the stream.
+            # Setting a timeout value, 'timeout_seconds', for the stream.
             # timeout value for connection to the server
             # If do not set then we will not able to stop immediately,
-            # becuase synchronus function watch.stream() will not come back
+            # because synchronous function watch.stream() will not come back
             # until catch any event on which it is waiting.
-            kwargs[K8SClientConst.TIMEOUT_SECONDS] = K8SClientConst.VAL_WATCH_TIMEOUT_DEFAULT
+            monitor_args['watch_args'][K8SClientConst.TIMEOUT_SECONDS] = wait_time
 
             # Get MessageBus producer object for all monitor threads
             producer = self._get_producer()
 
             # Change to multiprocessing
             # Creating NODE monitor object
-            node_monitor = ObjectMonitor(producer, K8SClientConst.NODE, **kwargs)
+            Log.info("Instantiating monitor for all the nodes in cluster.")
+            node_monitor = ObjectMonitor(producer, K8SClientConst.NODE, **monitor_args)
             self.monitors.append(node_monitor)
 
-            _, nodes_list = _conf_stor_search.get_cluster_cardinality()
-            if not nodes_list:
-                Log.warn(f"No nodes in the cluster to watch for nodes_list: {nodes_list}")
+            _, label_id_map, resource_id_map = _conf_stor_search.get_cluster_cardinality()
+            Log.debug(f"label id map: {label_id_map}, resource id map: {resource_id_map}")
+            if not label_id_map:
+                Log.warn(f"No pods ids found to monitor in pod id list: {label_id_map}")
             else:
-                Log.info(f"Starting watch for: nodes_list: {nodes_list}")
-            watcher_node_ids = ', '.join(node_id for node_id in nodes_list)
-            kwargs[K8SClientConst.LABEL_SELECTOR] = f'cortx.io/machine-id in ({watcher_node_ids})'
+                Log.info(f"Instantiating monitor for pods with 'cortx.io/machine-id' labels: {label_id_map.keys()}")
+                label_ids = ', '.join(label_id for label_id in label_id_map.keys())
+                monitor_args['watch_args'][K8SClientConst.LABEL_SELECTOR] = \
+                    f'{K8SEventsConst.LABEL_MACHINEID} in ({label_ids})'
+                monitor_args['resource_id_map'] = label_id_map
+
+                # Creating POD monitor object watching on label machine id
+                pod_monitor_for_machineids = ObjectMonitor(producer, K8SClientConst.POD, **monitor_args)
+                self.monitors.append(pod_monitor_for_machineids)
+
+            if not resource_id_map:
+                Log.warn(f"No pod names found to monitor in resource id map: {resource_id_map}")
```

Comment:
This should not be a warning since the identifiers are going to be present either in label_id or in resource_id
Same for the previous if not check

Reply:
Done, removed if not warn

Reply:
@mukhtar-inamdar , based on the discussion we had with Madhura, this comment is invalid.  since as discussed; both maps are always going to be present.
So you need to retain the warnings. (or have a common warning where 1 or both are none) Sorry abt the additional changes

Reply:
done